### PR TITLE
Add registries

### PR DESCRIPTION
**Pull request by adambkaplan**
Add MCO registries code to runtime-utils

/assign @gabemontero 

/cc @mtrmac

### CODE REVIEW

**Comment by adambkaplan** [comment]
/hold

Tests aren't running???

**Review by mtrmac — COMMENTED** [review]
LGTM.

**Comment by gabemontero** [comment]
/lgtm

in case getting the tests to run only require openshift/release changes and not new commits here

the MCO registries code copy looks good

we'll ultimately change https://github.com/openshift/openshift-controller-manager/blob/master/pkg/build/controller/build/build_controller.go#L2081 with a vendor bump on OCM for this repo

**Comment by mtrmac** [comment]
`make test` running locally does run the tests for me, FWIW.

**Comment by adambkaplan** [comment]
/hold

May not have picked up new jobs?

**Comment by mtrmac** [comment]
Thanks! Filed https://github.com/openshift/machine-config-operator/pull/1170 to use this in the MCO repo.

File: pkg/registries/doc.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Package registries contains shared infrastructure to build/update /etc/containers/registries.conf
+// based on cluster-wide configuration.
+//
+// NOTE: This has users outside of the machine-config-operator repository!
```
Comment (wking) [inline pkg/registries/doc.go:4]: Who are they?  I can't turn them up in [GitHub's search][1] or [GoDoc][2], although neither of those are exhaustive.  But the idea that we'll have multiple repositories calling `EditRegistriesConfig` freaks me out a little ;).

[1]: https://github.com/search?q=github.com%2Fopenshift%2Fruntime-utils%2Fpkg%2Fregistries&type=Code
[2]: https://godoc.org/github.com/openshift/runtime-utils/pkg/registries?importers

**Comment by adambkaplan** [comment]
@wking openshift-controller-manager and builder will use this. Working on this for the current sprint.

File: pkg/registries/doc.go (hunk 1 above)
Comment (mtrmac) [reply pkg/registries/doc.go:4]: - This was copy&pasted from MCO, and the wording above, warning people not to change the API unaware, is no longer really pertinent — it’s neither necessary to warn so strongly, nor it’s really clear why this should refer to the MCO repository at all.
- `EditRegistriesConfig` is a memory-only operation; it does not change any on-disk configuration itself, it’s up to the caller to read the created data, or to write it on disk somewhere and make it used. E.g. https://github.com/openshift/openshift-controller-manager/pull/19 creates a config map entry that is then mounted into a container. It’s not an automatic invitation for all of the callers to race each other overwriting node-level configuration files (this repo neither exactly prevents them from doing so, nor does it make it any easier).

**Pull request merged**